### PR TITLE
list active users to new allocation form

### LINKED ISSUE #180: Allow form of new allocation form to list all active users
If some user has active allocation and the admin tries to backfill allocations, the new allocation page does not show the user in the select field, we need to change to list all active users and just show some validation message if the admin is trying to create an allocation with overlap dates.

### PR DESCRIPTION
**Pull request by tonyaraujop**
Closes #180

User allocated:
image

New allocation with that user:
image

All allocations:
image

Error message if new allocation has overlapped dates
image

### CODE REVIEW

File: spec/queries/users_by_company_query_spec.rb [hunk 1]
```diff
@@ -18,17 +20,23 @@
 
   subject { UsersByCompanyQuery.new(company) }
 
-  describe '#not_allocated_including' do
-    context 'when passes user allocated' do
-      it 'return all users not allocated including user passed' do
-        result = subject.not_allocated_including(user_allocated)
+  describe '#active_engineers' do
+    context 'when users are engineers and actives' do
```
Comment (GabrielRMuller) [inline spec/queries/users_by_company_query_spec.rb:24]: `are active engineers` 💅

File: spec/features/admin/allocations_spec.rb [hunk 2]
```diff
@@ -67,8 +67,8 @@
 
       before { click_link 'Novo(a) Alocação' }
 
-      it 'should not render users allocated' do
-        expect(page).not_to have_text(user.name)
+      it 'should render users allocated' do
```
Comment (GabrielRMuller) [inline spec/features/admin/allocations_spec.rb:70]: I know you're just using the wording from the previous spec, but let's start migrating on to the correct spec wording and remove `should` (unnecessary when using `expect`).
```suggestion
      it 'renders the allocated users' do
```
See more: https://www.betterspecs.org/#expect

File: spec/queries/users_by_company_query_spec.rb [hunk 3]
```diff
@@ -18,17 +20,23 @@
...
+      it 'return all users' do
+        result = subject.active_engineers
         expect(result).to containing_exactly(user_not_allocated, user_allocated)
```
Comment (GabrielRMuller) [inline spec/queries/users_by_company_query_spec.rb:27]: not your code, but if you could change this to `expect(result).to contain_exactly` that'd be great ☕

File: spec/queries/users_by_company_query_spec.rb [hunk 4]
```diff
@@ -18,17 +20,23 @@
...
       end
     end
 
-    context 'when not passes user allocated' do
-      it 'return all users not allocated' do
-        expect(subject.not_allocated_including).to contain_exactly user_not_allocated
+    context 'when user it is not an engineer' do
```
Comment (GabrielRMuller) [inline spec/queries/users_by_company_query_spec.rb:31]:
```suggestion
    context 'when the user is not an engineer' do
```

File: spec/queries/users_by_company_query_spec.rb [hunk 5]
```diff
@@ -5,6 +5,8 @@
   let(:project) { create(:project, company: company) }
   let(:user_allocated) { create(:user, company: company) }
   let(:user_not_allocated) { create(:user, company: company) }
+  let(:not_engineer) { create(:user, occupation: 'administrative', company: company) }
```
Comment (GabrielRMuller) [inline spec/queries/users_by_company_query_spec.rb:8]:
```suggestion
  let(:non_engineer) { create(:user, occupation: 'administrative', company: company) }
```

File: spec/queries/users_by_company_query_spec.rb [hunk 6]
```diff
@@ -5,6 +5,8 @@
...
+  let(:not_active) { create(:user, company: company, active: false) }
```
Comment (GabrielRMuller) [inline spec/queries/users_by_company_query_spec.rb:9]:
```suggestion
  let(:inactive) { create(:user, company: company, active: false) }
```

**Review by GabrielRMuller — COMMENTED** [review]
code looks good! left some nits to improve specs readability

File: spec/queries/users_by_company_query_spec.rb (hunk 1 above)
Comment (tonyaraujop) [reply spec/queries/users_by_company_query_spec.rb:24]: Done!

File: spec/features/admin/allocations_spec.rb (hunk 2 above)
Comment (tonyaraujop) [reply spec/features/admin/allocations_spec.rb:70]: Ok.

**Review by GabrielRMuller — APPROVED** [review]

**Pull request merged**